backend/endpoints/messages.py

```diff
@@ -77,3 +78,92 @@ async def send_message(
+        status.HTTP_404_NOT_FOUND: {"description": "message not found"},
+        status.HTTP_403_FORBIDDEN: {
+            "description": "you are not authorized to edit this message"
+        },
+        status.HTTP_424_FAILED_DEPENDENCY: {"description": "message not edited"},
+    },
+)
+async def update_message(
+    request: MessageUpdate,
+    org_id: str,
+    room_id: str,
+    message_id: str,
+    background_tasks: BackgroundTasks,
+):
+    """
+    Update a message
+
+    Args:
+        request: Request object
+        org_id: A unique identifier of the organization.
+        room_id: A unique identifier of the room.
+        sender_id: A unique identifier of the sender.
+        message_id: A unique identifier of the message that is being edited.
+
+    Returns:
+        HTTP_200_OK {message updated successfully}:
+        A dict containing data about the message that was updated (response_output).
+            {
+                "room_id": "619e28c31a5f54782939d59a",
+                "message_id": "61bc5e5378fb01b18fac1426",
+                "sender_id": "619ba4671a5f54782939d385",
+                "text": "testing edits",
+                "edited_at": "2021-12-17 11:47:22.678046"
+            }
+
+    Raises:
+        HTTP_404_FAILED_DEPENDENCY: Message not found
+        HTTP_424_FAILED_DEPENDENCY: Message not edited
+        HTTP_403_FORBIDDEN: You are not authorized to edit this message
+    """
+    DB = DataStorage(org_id)
+    message = await get_mssg(org_id=org_id, room_id=room_id, message_id=message_id)
+
+    if not message:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Message not found"
+        )
+
+    payload = request.dict()
+    if message["sender_id"] != payload["sender_id"]:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="You are not authorized to edit this message",
+        )
+        
+    try:
+        edit_message = await DB.update(
+            MESSAGE_COLLECTION, document_id=message_id, data=payload
+        )
+        if edit_message:
+            data = {
+                "room_id": room_id,
+                "message_id": message_id,
+                "sender_id": payload["sender_id"],
+                "text": payload["text"],
+                "edited_at": payload["edited_at"],
+            }
+            background_tasks.add_task(
+                centrifugo_client.publish, room_id, Events.MESSAGE_UPDATE, data
+            )  # publish to centrifugo in the background
+            return JSONResponse(
+                content=ResponseModel.success(
+                    data=data, message="message edited"
+                ),
+                status_code=status.HTTP_200_OK,
+            )
+        raise HTTPException(
+            status_code=status.HTTP_424_FAILED_DEPENDENCY,
+            detail={"message not edited": edit_message},
+        )
+    except Exception as e:
```

Comment:
is there any error the try block will get.
Also if an error is caught what response will be returned ?

Reply:
Yes it would raise a failed dependency error. I have made it more explicit though

Reply:
Ohh
Actually No.
It would return the raised http exception (failed dependency) as a response to the endpoint and wouldn't actually cause an program exception to the code itself.